### PR TITLE
#8788: Support fp32 dest acc en to moreh_linear OP

### PR DESCRIPTION
**Pull request by dongjin-na**
This PR adds the DeviceComputeKernelConfig parameter to operations like these functions.
- moreh_linear
- moreh_linear_backward
- moreh_matmul
- moreh_matmul_backward
- moreh_bias_add_backward 

This change will allow us to provide kernel configuration settings optionally during function calls.

### CODE REVIEW

**Review by hschoi4448 — APPROVED** [review]

File: tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_matmul.py [hunk 1]
```diff
@@ -350,5 +365,55 @@ def test_moreh_matmul_wo_output(params, device):
 def test_moreh_matmul_enable_cache(params, device, use_program_cache):
     torch.manual_seed(3072)
     for i in range(2):
-        passing = moreh_matmul(params, True, device)
+        passing = moreh_matmul(params, True, None, device)
         assert passing
```
Comment (TT-BrianLiu) [inline tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_matmul.py:369]: Same comment as https://github.com/tenstorrent/tt-metal/pull/8966#discussion_r1622522504 about testing program cache.

Comment (dongjin-na) [reply tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_matmul.py]: Dear @TT-BrianLiu ,
I added a test case to check the program cache. Would you please review it?

**Comment by sangwon-chae** [comment]
@davorchap @TT-BrianLiu Would you please review this PR?

**Comment by davorchap** [comment]
> 
> 
> This PR adds the DeviceComputeKernelConfig parameter to operations like these functions.
> 
> - moreh_linear
> 
> - moreh_linear_backward
> 
> - moreh_matmul
> 
> - moreh_matmul_backward
> 
> - moreh_bias_add_backward 
> 
> 
> 
> This change will allow us to provide kernel configuration settings optionally during function calls.
> 
> 
> 
> 

@yugaoTT please take a look

**Review by yugaoTT — APPROVED** [review]

**Comment by dongjin-na** [comment]
Dear code owners @TT-BrianLiu, @tt-aho, @mywoodstock,

Would you please review this PR?

(CC @sangwon-chae)

**Review by mywoodstock — APPROVED** [review]

**Pull request merged**